plugin/pkg/admission/security/podsecuritypolicy/admission.go

```diff
@@ -135,6 +135,79 @@ func (c *podSecurityPolicyPlugin) Admit(a admission.Attributes) error {
 		return nil
 	}
 
+	if _, ok := pod.ObjectMeta.Annotations[psputil.ValidatedPSPAnnotation]; ok {
+		// the validator will verify that the given PSP name actually matches.
+		return nil
+	}
+
+	// TODO(liggitt): allow spec mutation during initializing updates?
+	specMutationAllowed := a.GetOperation() == admission.Create
+
+	// compute the context. If the current security context is valid, this call won't change it.
+	allowedPod, pspName, validationErrs, err := c.computeSecurityContext(a, pod, specMutationAllowed)
+	if err != nil {
+		return admission.NewForbidden(a, err)
+	}
+	if allowedPod != nil {
+		*pod = *allowedPod
+		// annotate and accept the pod
+		glog.V(4).Infof("pod %s (generate: %s) in namespace %s validated against provider %s", pod.Name, pod.GenerateName, a.GetNamespace(), pspName)
+		if pod.ObjectMeta.Annotations == nil {
+			pod.ObjectMeta.Annotations = map[string]string{}
+		}
+		// set annotation to mark this as passed. Note, that the actual value is not important, the
+		// validating PSP might even change lateron. Also not that pspName can be the empty string
+		// if failOnNoPolicies is false.
+		// TODO: if failOnNoPolicies is toggled from false to true, we will never update the annotation anymore. Is this desired?
+		pod.ObjectMeta.Annotations[psputil.ValidatedPSPAnnotation] = pspName
+		return nil
+	}
+
+	// we didn't validate against any provider, reject the pod and give the errors for each attempt
+	glog.V(4).Infof("unable to validate pod %s (generate: %s) in namespace %s against any pod security policy: %v", pod.Name, pod.GenerateName, a.GetNamespace(), validationErrs)
+	return admission.NewForbidden(a, fmt.Errorf("unable to validate against any pod security policy: %v", validationErrs))
+}
+
+func (c *podSecurityPolicyPlugin) ValidatingAdmit(a admission.Attributes) error {
```

Comment:
This ended up as validate, right?  Add type assertions please.

Reply:
right, missed that after rebasing

Reply:
done